core/solver/bicg.cpp

```diff
@@ -129,6 +129,8 @@ void Bicg<ValueType>::apply_dense_impl(const matrix::Dense<ValueType>* dense_b,
 
     auto exec = this->get_executor();
 
+    Array<char> reduction_tmp{exec};
```

Comment:
I guess the idea is to move this to one of the base classes in the value semantics (solver base class) PRs ?